src/platform/datapath_xplat.c

```diff
@@ -157,11 +160,19 @@ CxPlatSocketCreateUdp(
             QuicTraceLogVerbose(
                 RawSockCreateFail,
                 "[sock] Failed to create raw socket, status:%d", Status);
-            if (Datapath->UseTcp) {
+            BOOLEAN IsWildcardAddr = Config->LocalAddress == NULL || QuicAddrIsWildCard(Config->LocalAddress);
+            if (IsWildcardAddr && TryCount < 1000 && (Config->Flags & CXPLAT_SOCKET_FLAG_QTIP)) {
                 CxPlatSocketDelete(*NewSocket);
+                TryCount += 1;
+                goto Retry;
```

Comment:
I would avoid this goto.
The semantic you want here is a loop - and if the loop is hard to write, it probably means its body should be moved to a subfunction.

I think the logic if complicated enough it deserves a comment about when/why we are sometimes fine with not creating the raw socket. For instance, it might be correct and intentional, but I am surprised to see we are going to try 1000 times to create a pair of sockets to eventually say "eh, I don't need QTIP anyway, lets succeed", and a comment would help understand why :)

Reply:
Addressed offline + committed fix.